specs/tilebag_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'minitest/autorun'
+require 'minitest/reporters'
+require 'minitest/skip_dsl'
+
+require_relative '../lib/player'
+require_relative '../lib/scoring'
+require_relative '../lib/tilebag'
+
+Minitest::Reporters.use! Minitest::Reporters::SpecReporter.new
+
+describe 'TileBag' do
+
+  before do
+    @bag = Scrabble::TileBag.new
+  end
+
+  describe 'initialize' do
+    it 'has a collection of all tiles' do
+      new_bag = Scrabble::TileBag.new
+      new_bag.must_be_instance_of Scrabble::TileBag
+      new_bag.tilebag.length.must_equal 26
```

Comment:
You don't need to test the internal structure of the `Tilebag` as a user you don't care that the `TileBag` has a hash attribute, in fact that shouldn't even be accessible to you.  Instead you should test the `tiles_remaining` method and the other methods listed in the requirements.